cpp/src/gandiva/precompiled/time.cc

```diff
@@ -691,17 +728,41 @@ timestamp castTIMESTAMP_date64(date64 date_in_millis) { return date_in_millis; }
 
 char* castVARCHAR_timestamp_int64(int64 context, timestamp in, int64 length,
                                   int32* out_len) {
-  std::string timestamp_str = std::to_string(in);
+  int64 year = extractYear_timestamp(in);
+  int64 month = extractMonth_timestamp(in);
+  int64 day = extractDay_timestamp(in);
+  int64 hour = extractHour_timestamp(in);
+  int64 minute = extractMinute_timestamp(in);
+  int64 second = extractSecond_timestamp(in);
+  int64 millis = in % MILLIS_IN_SEC;
+
+  // format to yyyy-MM-dd hh:mm:ss.mil
+  std::stringstream s;
+  s << ensure_four_digits(year) << "-" << ensure_two_digits(month) << "-"
+    << ensure_two_digits(day) << " " << ensure_two_digits(hour) << ":"
+    << ensure_two_digits(minute) << ":" << ensure_two_digits(second) << "."
+    << ensure_three_digits(millis);
+
+  std::string timestamp_str = s.str();
   *out_len = static_cast<int32>(length);
   int32 timestamp_str_len = static_cast<int32>(timestamp_str.length());
   if (length > timestamp_str_len) {
     *out_len = timestamp_str_len;
   }
+  if (*out_len <= 0) {
+    if (*out_len < 0) {
+      gdv_fn_context_set_error_msg(context, "Length of output string cannot be negative");
+    }
+    *out_len = 0;
+    const char* empty_str = "";
+    return const_cast<char*>(empty_str);
+  }
   char* ret = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, *out_len));
   if (ret == nullptr) {
     gdv_fn_context_set_error_msg(context, "Could not allocate memory for output string");
     *out_len = 0;
-    return nullptr;
+    const char* empty_str = "";
+    return const_cast<char*>(empty_str);
```

Comment:
simply, return "" should work too ? or, add one static char *EMPTY_STRING = "";

Reply:
changed function return type to const char*. Thanks